command/server.go

```diff
@@ -1279,7 +1279,7 @@ CLUSTER_SYNTHESIS_COMPLETE:
 
 	// If ServiceRegistration is configured, then the backend must support HA
 	isBackendHA := coreConfig.HAPhysical != nil && coreConfig.HAPhysical.HAEnabled()
-	if (coreConfig.ServiceRegistration != nil) && !isBackendHA {
+	if !c.flagDev && (coreConfig.ServiceRegistration != nil) && !isBackendHA {
```

Comment:
Is there a technical reason we don't support dev mode?  Do we have any other configurations that don't work in dev mode?

Reply:
@mjarmy or @briankassouf do you happen to know the answer to this?

@jasonodonnell , I don't know the original reason for this, but I _do_ know that Consul currently also has service registration and it presently is only available in HA. Right [here](https://github.com/hashicorp/vault/blob/master/serviceregistration/consul/consul_service_registration.go#L530), it builds how it's going to advertise itself based on whether it's a performance standby. If we were going to make service registration available in wider circumstances, I'm not sure if that could negatively impact Consul from a backwards-compatibility standpoint.

Reply:
Actually, when I re-read this, I see the question is about _dev mode_, not HA. I just was thinking about HA because we'd talked about it previously.

In response to the _actual question_ :-), I'd bet that we weren't allowed to run service registrations unless in HA mode because some of the service registration notifications only work with HA. However, this made it difficult to do just little quick tests that didn't pertain to the HA notifications. So I added a flag to allow us to test service registration if the `-dev` flag is included just to make my life a little easier. With this exception added here, I can just do a quick test without running in a clustered mode, or even if I've dropped one single Vault instance into a container. I figured that since it doesn't effect end users, should be fine.